keras_nlp/models/bert.py

```diff
@@ -0,0 +1,287 @@
+        self.initializer_fn = initializer_fn
+        self.vocab_size = vocab_size
+        self.hidden_size = hidden_size
+        self.num_layers = num_layers
+        self.num_heads = num_heads
+        self.max_sequence_length = max_sequence_length
+        self.num_segments = num_segments
+        self.intermediate_dim = intermediate_dim
+        self.dropout = dropout
+
+    def get_embedding_table(self):
+        return self.token_embedding.embeddings
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "vocab_size": self.vocab_size,
+                "hidden_size": self.hidden_size,
+                "num_layers": self.num_layers,
+                "num_heads": self.num_heads,
+                "max_sequence_length": self.max_sequence_length,
+                "num_segments": self.num_segments,
+                "intermediate_dim": self.intermediate_dim,
+                "dropout": self.dropout,
+            }
+        )
+        return config
+
+
+class BertClassifier(keras.Model):
+    """
+    Adds a classification head to a Bert encoder model.
+
+    Args:
+        encoder: A `Bert` Model to encode inputs.
+        num_classes: Number of classes to predict.
+        kernel_initializer: Initializer for the `kernel` weights matrix.
+        bias_initializer: Initializer for the bias vector.
+
+    Example usage:
+    # Randomly initialized Bert encoder
+    encoder = keras_nlp.models.Bert(
+        vocab_size=30522,
+        num_layers=12,
+        hidden_size=768,
+        num_heads=12,
+        intermediate_dim=3072,
+        dropout=0.1,
+        max_sequence_length=12
+    )
+
+     # Call classifier on the inputs.
+    input_data = {
+        "input_ids": tf.random.uniform(
+            shape=(1, 12), dtype=tf.int64, maxval=30522),
+        "segment_ids": tf.constant(
+            [0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)),
+        "input_mask": tf.constant(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)),
+    }
+    classifier = keras_nlp.models.BertClassifier(encoder, 4)
+    logits = classifier(input_data)
+    """
+
+    def __init__(
+        self,
+        encoder,
+        num_classes,
+        kernel_initializer="glorot_uniform",
+        bias_initializer="zeros",
+        **kwargs,
+    ):
+        inputs = encoder.input
+        pooled = encoder(inputs)["pooled_output"]
+        outputs = keras.layers.Dense(
+            num_classes,
+            kernel_initializer=kernel_initializer,
+            bias_initializer=bias_initializer,
+            name="logits",
+        )(pooled)
+        # Instantiate using Functional API Model constructor
+        super().__init__(inputs=inputs, outputs=outputs, **kwargs)
+        # All references to `self` below this line
+        self.encoder = encoder
+        self.num_classes = num_classes
+
+
+def BertBase(**kwargs):
+    """
+    Factory for Bert using "Base" architecture.
```

Comment:
I don't think we should refer to these as factories. To a user this will just appear to be a model object, the fact that we are doing this via a function (and not a subclass), is an implementation detail we can change as we would like.

No current CV applications document themselves as a factory.